### DISCUSSION TITLE
Can you do a comparison to JLibrosa?

### DISCUSSION THREAD

**Discussion by mavavilj** [psambit9791/jDSP#51] [General]
I've been looking for a BSD/MIT replacement for TarsosDSP, and I came across this. It doesn't have all the stuff, but I was then interested on maybe contributing them.

But I was also interested, can you do a comparison to JLibrosa (https://github.com/Subtitle-Synchronizer/jlibrosa)?

**Comment by psambit9791** [comment]
## Comparison with TarsosDSP

JDSP is fundamentally a signal processing library. On the other hand, TarsosDSP/JLibrosa are both audio processing libraries. This implies that JDSP serves a different set of requirements than audio processing libraries. Audio processing uses a set of higher order functions which are used specifically for "auditory signals". Signal processing tools can be much lower level and can be used on any signals ranging from accelerometer/gyroscope signals  to audio. 

JDSP provides tools which can be used to create audio libraries, and as of now, we are trying to expand the toolkit to include some audio processing functions like Silence Detection and PSOLA. Since TarsosDSP is primarily an audio processing library, the functionalities available are different from those available on JDSP.

## Comparison with JLibrosa

#### Common Functionalities between JLibrosa & JDSP

- Reading of WAV files
- Implementation of STFT and iSTFT

#### Additional Functionalities in JLibrosa

- MelSpectrogram Value Generation
- MFCC Feature Value Generation


Hope this helps.
